pkg/protobuf/comparer_test.go

```diff
@@ -0,0 +1,532 @@
+package protobuf_test
+
+import (
+	"testing"
+
+	"github.com/syncromatics/proto-schema-registry/pkg/protobuf"
+
+	"gotest.tools/assert"
+)
+
+func Test_Comparer_WithFieldTypeChange(t *testing.T) {
+	current := `syntax = "proto3";
+	package gen;
+
+	message first {
+		string one = 1;
+		int32 two = 2;
+	}
+
+	message record {
+		first first_message = 1;
+	}
+	`
+
+	new := `syntax = "proto3";
+	package gen;
+
+	message first {
+		int32 one = 1;
+		int32 two = 2;
+	}
+
+	message record {
+		first first_message = 1;
+	}
+	`
+
+	ok, problems, err := protobuf.CheckForBreakingChanges([]byte(current), []byte(new))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, false, ok)
+	assert.DeepEqual(t, []string{
+		"message 'first' had field type 'string' changed to 'int32' at sequence '1'",
+	}, problems)
+}
+
+func Test_Comparer_WithFieldRemove(t *testing.T) {
```

Comment:
Why is this a failure case? Per proto3 language guide:

> If the field is not used by anyone and you want to prevent new users from using it, consider replacing the field declaration with a reserved statement.

It's not unreasonable to remove or cease using a field number after deprecating it.

Reply:
From where my train of though was: If we allow the removal and something is added in it's place later we would have to go through all versions to ensure the type didn't change